ssl/ssl_sess.c

```diff
@@ -161,12 +161,12 @@ SSL_SESSION *SSL_get1_session(SSL *ssl)
      * somebody doesn't free ssl->session between when we check it's non-null
      * and when we up the reference count.
      */
-    CRYPTO_w_lock(CRYPTO_LOCK_SSL_SESSION);
+    CRYPTO_THREAD_read_lock(ssl->lock);
```

Comment:
The original was a write lock, but I thought read was more appropriate since the `ssl` variable is not modified. It's unlikely to make any difference though.